### DISCUSSION TITLE
Modal does not close, when I wanna save and close.

### DISCUSSION THREAD

**Discussion by anotherMichaelMaier** [neuefische/questions#518] [Web] [answered]
Hey everyone,

I made a edit modal, it works as intended. The only problem is I can only close it via the X in the top right, but I want it also to close when the User want to save the changes made.
I tried a lot already but I just don't know how to get the `onClose` work as I need it to.

Here is my PR [https://github.com/anotherMichaelMaier/capstone-project/pull/18](url)

I appreciate your help and afford! 
Thanks in advance.

**Comment by marcterre** [comment, marked as answer]
Hey Michael, your link leads me to a 404.
In your code you got the  ```const [showModal, setShowModal] = useState(false);```. If you click on your edit button you set the state true and the modal will be displayed. What worked for me is to ```setShowModal(false)``` when submitting.
You could write 
```js
 <ModalEditCard
       id={id}
       name={name}
       note={note}
       time={time}
       tasks={tasks}
       updateTask={() => {updateTask();
       setShowModal(false}}
/>
```
Hope this works

**Reply by mbosselmann** [reply]
Hey you two,
Marc's solution is a good working solution. 👍 You could also think about setting the prop `onClose={() => setShowModal(false)}` as you have done for your `button` and the `Modal` in the same file:

```js
  <ModalEditCard
       id={id}
       name={name}
       note={note}
       time={time}
       tasks={tasks}
       updateTask={updateTask}
       onClose={() => setShowModal(false)}
  />
```
Then you can add `onClose` to your `handleSubmit` function in your `ModelEditCard` and it would work the same. :)

I hope our ideas does help you in moving further! 💪
